Framework/src/CheckRunner.cxx

```diff
@@ -91,11 +92,25 @@ std::string CheckRunner::createCheckRunnerName(std::vector<Check> checks)
     // If single check, use the check name
     name += checks[0].getName();
   } else {
-    std::default_random_engine generator;
-    std::uniform_int_distribution<int> distribution(0, alphanumeric.size());
+    std::string hash_string = "";
+    std::vector<std::string> names;
+    // Fill vector with check names
+    for (auto& c : checks) {
+      names.push_back(c.getName());
+    }
+    // Be sure that after configuration shuffle, the name will be the same
+    std::sort(names.begin(), names.end());
+
+    // Create a single string and hash it
+    for (auto& n : names) {
+      hash_string += n;
+    }
+    std::size_t num = std::hash<std::string>{}(hash_string);
```

Comment:
Does std::hash guarantee determinism and is platform independent?
I am worried that we might stumble upon some problems when integrating with the Control system. It requires separate workflow specifications uploaded on a repository, like here https://github.com/AliceO2Group/ControlWorkflows/blob/master/workflows/readout-qc-1.yaml . If the generated device name doesn't match a specification, because the it was created on a different machine, different day or with a different compiler version, then we might have problems running such a workflow.
That being said, this is still of course better than using pseudo-random values, thanks for that!

Reply:
> returning always the same value for the same argument (for a given execution of a program)

So I guess it's a valid concern and the implementation might differ between the compilers/stdlib providers.
I will look into determinism guaranteed functions (sha1).

Reply:
Does it prevent merging or will you make a new PR in the future ?

Reply:
That will force us to change the checker name again in the workflow specification for AliECS, so I would rather avoid doing this often until it is easier to do.

Reply:
I understand.

Reply:
I was looking into this, but there is no standard way of hashing in C++, or at least I didn't find it. It will require a external library. The most common library would be OpenSSL. Could I assume that it will be preinstalled on the system or by other means and use the functionality from this library?

Reply:
Maybe a very stupid hash would be enough? For example casting all characters to numbers and adding them together. What do you think?

Reply:
I was considering this, but the problem with the addition arises if we use the same set of chars or it has other numeric correlation, e.g. Check2 == 2Check or AE == BD. But thinking of it now I could try the xor operation on it and this could work.
Other solutions that I consider are a header-only library, although I am not sure about the licencing, or based on the random idea, provide an explicit seed number, which should provide the same results even after restart. It could though have the same problem as the proposed solution in this PR.

Reply:
Xor doesn't care about the order as far as I understand, so Check2 == 2Check would still result in the same hash. Then how about something like this:
```
short hash = 0;
size_t i = 1;
for (const auto& name : names) {
  for (const char c : name) {
    hash += static_cast<short>(c) * i++;
  }
}
```
Though this still might fail for AE and CD. A xor could be added somewhere to fix it.

Reply:
As a next proposal, I found a quite simple hashing algorithm: BSD checksum algorithm.

Reply:
That one looks nice! I gave it a try, but so far testCheckWorkflow is failing - a segfault, but probably on something with setCustomParameter. Could you have a look?

Reply:
I was not able to reproduce the segfault. I merged master into the branch and for me, all the test are still working. Could you test it now after the merge? If you still encourage the problem, please send me the stacktrace and if possible used version via mail.

Reply:
I've retried after you've merged, now it runs nicely. No idea what was the problem. Thanks for those improvements!